### PR TITLE
 Routes & Controller responses updated

### PR DESCRIPTION
**Pull request by mggaray**
please review and approve if there are no problems

### CODE REVIEW

File: src/controllers/foods.js [hunk 1]
```diff
@@ -25,12 +25,24 @@ export class FoodsController extends BaseSQLController {
...
+					success: false,
+					message: error.message,
+					httpStatusCode: 500,
+					response: error.error
+				})
 		);
 	}
 
-	async getFoodByNameMethods(req, res) {
+	async getByFoodName(req, res) {
```
Comment (julian-er) [inline src/controllers/foods.js]: Why should you change this name? this is a method to query by name but not return an hhtp response.
In fact, you should have a getFoodByName function that uses this method and then returns a response like in the user's handlers.

File: src/controllers/allergies.js [hunk 2]
```diff
@@ -16,8 +16,44 @@ export class AllergiesController extends BaseSQLController {
...
+				res.status(500).json({
+					success: false,
+					message: error.message,
+					httpStatusCode: 500,
+					response: error.error
+				})
+		);
+	}
+
+	getAllegyById(req, res) {
```
Comment (julian-er) [inline src/controllers/allergies.js]: Please correct the typo, you are trying to use this function in the router because of this it will not work 😞

Comment (julian-er) [reply src/controllers/allergies.js]: Also... please you can fix the typo on line 10 too?

File: src/controllers/allergies.js [hunk 3]
```diff
@@ -27,7 +63,7 @@ export class AllergiesController extends BaseSQLController {
 	 * @param {Response} res The Express response
 	 */
 
-	async getByAllergyNameMethod(req, res) {
+	async getByAllergyName(req, res) {
```
Comment (julian-er) [inline src/controllers/allergies.js]: Why should you change this name? this is a method to query by name but not return an HTTP response.
In fact, you should have a getByAllergyName function that uses this method and then returns a response like in the user's handlers.

File: src/routes/allergies.js [hunk 4]
```diff
@@ -7,9 +7,10 @@ import { AllergiesController } from '../controllers/index.js';
  */
 export function setAllergiesRoutes(router) {
 	const controller = new AllergiesController();
-
 	router.get('/allergies', (req, res) => controller.getAllergies(req, res));
+	router.get('/allergies/:id', (req, res) => controller.getAllergyById(req, res));
```
Comment (julian-er) [inline src/routes/allergies.js:11]: Fix the typo on the allergies controller and this will work

**Comment by julian-er** [comment]
I can't approve this merge, for now. We have some issues. Please resolve this and let me know! 😄

File: src/routes/allergies.js (hunk 4 above)
Comment (julian-er) [reply src/routes/allergies.js:11]: Test review

**Review by julian-er — CHANGES_REQUESTED** [review]

File: src/controllers/allergies.js (hunk 2 above)
Comment (mggaray) [reply src/controllers/allergies.js]: resolved

File: src/routes/allergies.js (hunk 4 above)
Comment (mggaray) [reply src/routes/allergies.js:11]: resolved

File: src/controllers/foods.js (hunk 1 above)
Comment (mggaray) [reply src/controllers/foods.js]: fixed

File: src/controllers/allergies.js (hunk 3 above)
Comment (mggaray) [reply src/controllers/allergies.js]: done

**Comment by julian-er** [comment]
@mggaray the branch has conflicts, maybe you need to try marge main first

**Review by julian-er — APPROVED** [review]
Great !

**Review by AdanVolken — APPROVED** [review]
good job !!

**Pull request merged**